spec/ddtrace/context_provider_spec.rb

```diff
@@ -18,5 +26,31 @@
         expect(thread_local_context2.local).to eq(local_context2)
       end
     end
+
+    context 'in another thread' do
+      it 'create one thread-local variable per thread' do
+        context = thread_local_context.local
+
+        Thread.new do
+          expect { @thread_context = thread_local_context.local }.to change { Thread.current.keys.size }.by(1)
```

Comment:
is this asserting that `Thread.current.keys.size` changes from `0` to `1` or `1` to `2`?

Reply:
We should probably be more explicit here.

Reply:
It asserts that it increases by 1, given we don't know what's already in `Thread.current.keys`.

For example, in my terminal every thread starts with a few goodies:
```ruby
[1] pry(main)> Thread.current.keys
=> [:pry_critical_section, :__pry_frame_managers__, :__pry__]
```

I could make it more strict but it will probably just make it more brittle, with no further guarantee of correctness.

Reply:
maybe we should count based on keys that start with `datadog_context_` ?

Reply:
I'm not filtering only by `datadog_context_`, also changed the assertion to be more strict.